uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Mail/Template/BaseRequestNotification.php

```diff
@@ -47,20 +47,29 @@ public function __construct(LeaveRequestCommentService $leaveRequestComment) {
    * @return array
    */
   public function getTemplateParameters(LeaveRequest $leaveRequest) {
+    $calculationUnitName = $this->getAbsenceTypeCalculationUnitName($leaveRequest);
     $templateParameters =  [
       'leaveComments' => $this->getLeaveComments($leaveRequest),
       'leaveFiles' => $this->getAttachments($leaveRequest),
       'fromDate' => $leaveRequest->from_date,
       'toDate' => $leaveRequest->to_date,
-      'fromDateType' => $this->getLeaveRequestDayTypeLabel($leaveRequest->from_date_type),
-      'toDateType' => $this->getLeaveRequestDayTypeLabel($leaveRequest->to_date_type),
       'leaveStatus' => $this->getLeaveRequestStatusLabel($leaveRequest->status_id),
       'leaveRequest' => $leaveRequest,
       'absenceTypeName' => $this->getAbsenceTypeName($leaveRequest),
       'currentDateTime' => new DateTime(),
-      'calculationUnitName' => $this->getAbsenceTypeCalculationUnitName($leaveRequest)
+      'calculationUnitName' => $calculationUnitName
     ];
 
+    if($calculationUnitName == 'days') {
+      $dateTypeLabels = [
+        'fromDateType' => $this->getLeaveRequestDayTypeLabel($leaveRequest->from_date_type),
+        'toDateType' => $this->getLeaveRequestDayTypeLabel($leaveRequest->to_date_type),
+      ];
+
+      $templateParameters = array_merge($templateParameters, $dateTypeLabels);
```

Comment:
Instead of using array_merge, you just add them to the array:

```php
$templateParameters['fromDateType'] = $this->getLeaveRequestDayTypeLabel($leaveRequest->from_date_type);
$templateParameters['toDateType'] = $this->getLeaveRequestDayTypeLabel($leaveRequest->to_date_type);
```